1_download_dialogs_data.py

```diff
@@ -1,300 +1,110 @@
-
-        dialog.append(
-            {
-                "id": m.id,
-                "date": m.date,
-                "from_id": m.from_id,
-                "to_id": msg_attrs["to_id"],
-                "fwd_from": m.fwd_from,
-                "message": msg_attrs["message"],
-                "type": msg_attrs["type"],
-                "duration": msg_attrs["duration"],
-                "reactions": msg_reactions,
-            }
-        )
-
-    dialog_file_path = os.path.join(config["dialogs_data_folder"], f"{str(id)}.csv")
-
-    df = pd.DataFrame(dialog)
-    df.to_csv(dialog_file_path)
+    return [
+        dialog
+        for dialog in dialog_list
+        if is_dialog_type_accepted[dialog["type"]] and condition(dialog)
+    ]
 
 
 if __name__ == "__main__":
+    print("start downloading dialogs...")
+
     args = init_args()
 
-    CONFIG_PATH = args.config_path
-    MSG_LIMIT = msg_limit_input_handler(args.dialog_msg_limit)
+    MSG_LIMIT = 100_000_000 if args.dialog_msg_limit == -1 else args.dialog_msg_limit
     SESSION_NAME = args.session_name
-    DEBUG_MODE = args.debug_mode
 
     is_dialog_type_accepted = {
-        "Private dialog": not args.skip_private,
-        "Group": not args.skip_groups,
-        "Channel": not args.skip_channels,
+        DialogType.PRIVATE: not args.skip_private,
+        DialogType.GROUP: not args.skip_groups,
+        DialogType.CHANNEL: not args.skip_channels,
     }
+    print(f"dialog types to download: {is_dialog_type_accepted}")
 
-    config = init_config(CONFIG_PATH)
-    dialogs_list = read_dialogs(config["dialogs_list_folder"])
-    client = telethon.TelegramClient(
-        SESSION_NAME,
-        config["api_id"],
-        config["api_hash"],
-        system_version="4.16.30-vxCUSTOM",
+    dialog_reader = create_json_dialog_reader_writer()
+    dialogs = dialog_reader.read_all_dialogs()
+    print(f"total dialogs: {len(dialogs)}")
+    filtered_dialogs = filter_input_dialogs(
+        args.dialog_ids, is_dialog_type_accepted, dialogs
     )
+    print(f"total filtered dialogs: {len(filtered_dialogs)}")
 
-    DIALOGS_ID = dialogs_id_input_handler(
-        args.dialogs_ids, is_dialog_type_accepted, dialogs_list
-    )
-
-    # Dialogs are the "conversations you have open".
-    # This method returns a list of Dialog, which
-    # has the .entity attribute and other information.
-    # dialogs = client.get_dialogs()
-
-    if DEBUG_MODE:
-        logging.basicConfig(level=logging.DEBUG)
-
-    if not os.path.exists(config["dialogs_data_folder"]):
-        os.mkdir(config["dialogs_data_folder"])
-
-    for id in DIALOGS_ID:
-        print(f"Loading dialog #{id}")
-
-        with client:
-            client.loop.run_until_complete(
-                download_dialog(client, id, MSG_LIMIT, config)
+    client = create_telegram_client(SESSION_NAME)
+    print("downloading dialogs...")
+    with client:
+        try:
+            with client.takeout(
```

Comment:
make this hardcoded list of arguments configurable (move to config/env file), naming should have same prefix for a group of this arguments